rpc.go

```diff
@@ -65,11 +74,16 @@ func New(cfg Config) RPC {
 	if cfg.Name == "" {
 		panic("a service name is required")
 	}
+
+	filters := []transport.Filter(cfg.Filters)
+	interceptors := []transport.Interceptor(cfg.Interceptors)
 	return rpc{
-		Name:      cfg.Name,
-		Registry:  transport.NewMapRegistry(cfg.Name),
-		Inbounds:  cfg.Inbounds,
-		Outbounds: cfg.Outbounds,
+		Name:        cfg.Name,
+		Registry:    transport.NewMapRegistry(cfg.Name),
+		Inbounds:    cfg.Inbounds,
+		Outbounds:   cfg.Outbounds,
+		Filter:      transport.ChainFilters(filters...),
+		Interceptor: transport.ChainInterceptors(interceptors...),
```

Comment:
Also, Java takes a list of interceptors and filters, but Go just takes 1 and used the ChanFilter/Interceptor.

I think this is probably the most extensible/composable way (rpc object taking a single interceptor/filter) - @junchaowu think it would be hard to get Java acting like this?